csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -795,10 +807,23 @@ public Settings(bool formatDefaultValues) : this(formatDefaultValues, TypeRegist
             /// </summary>
             /// <param name="formatDefaultValues"><c>true</c> if default values (0, empty strings etc) should be formatted; <c>false</c> otherwise.</param>
             /// <param name="typeRegistry">The <see cref="TypeRegistry"/> to use when formatting <see cref="Any"/> messages.</param>
-            public Settings(bool formatDefaultValues, TypeRegistry typeRegistry)
+            public Settings(bool formatDefaultValues, TypeRegistry typeRegistry) : this(formatDefaultValues, typeRegistry, false)
+            {
+            }
+
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object with the specified parameters.
+            /// </summary>
+            /// <param name="formatDefaultValues"><c>true</c> if default values (0, empty strings etc) should be formatted; <c>false</c> otherwise.</param>
+            /// <param name="typeRegistry">The <see cref="TypeRegistry"/> to use when formatting <see cref="Any"/> messages. TypeRegistry.Empty will be used if it is null.</param>
+            /// <param name="formatEnumsAsIntegers"><c>true</c> format the enums as ints; <c>false</c> otherwise.</param>
+            public Settings(bool formatDefaultValues = false,
```

Comment:
I've been thinking about this, and we end up with a problem around optional parameters. Currently, if someone calls

```csharp
new Settings(formatEnumsAsIntegers: true)
```

then that's fine - but if we add another constructor, overload resolution will fail.

Basically, this is all slightly messy, but it might be worth not having defaults, but have "With" methods instead, so we have:

- 3x constructor (and more added as required), no optional parameters
- One "With" method per setting, e.g. WithFormatDefaultValues, WithTypeRegistry etc.

For the sake of avoiding dragging this PR out too much, if you *just* remove the defaults here, I can mail another PR to add the With methods. How does that sound?